ProjectDataCore.Components/Nav/EditModuleDisplay.razor

```diff
@@ -0,0 +1,28 @@
+﻿@using ProjectDataCore.Data.Structures.Nav
+@using ProjectDataCore.Data.Services.Nav
+@inject INavModuleService _navModuleService
+<li>
+    <span @ondblclick="()=>OnDblClick(Module)">@Module.DisplayName | <a class="underline text-sky-500" href="@Module.Href">@Module.Href</a></span>
+        <ul class="">
+            @foreach (NavModule m in Module.SubModules)
+            {
+                   <EditModuleDisplay Module="@m"/>
+            }
+            <li class="last"><span @onclick="()=>OnLeftClick(Module)">Add +</span></li>
+        </ul>
+</li>
+
+@code {
+    [Parameter] public NavModule Module { get; set; }
+
+    void OnDblClick(NavModule m)
+    {
+        _navModuleService.OnDblClickTrigger(this, m);
+    }
+
+    void OnLeftClick(NavModule m)
+    {
+        _navModuleService.OnLeftClickTrigger(this, m);
+    }
+
+}
```

Comment:
All code should be in code-behind files. You can use quick actions and refactoring to move the `@code` block into one of those files automatically.